point_pattern.py

```diff
@@ -119,11 +138,20 @@ def average_nearest_neighbor_distance(points):
      Measure of Spatial Relationships in Populations. Ecology. 35(4)
      p. 445-453.
     """
-    mean_d = 0
-
+    nearest = []
+    for i, point in enumerate(points):
+         nearest.append(None)
+         for point2 in points:
+             if point is not point2:
+                 dist = euclidean_distance(point, point2)
```

Comment:
What if the two points are coincident?  Check out `enumerate` to track the position (index) in both lists.  if the index is equal, then the points are identical.  Otherwise, they might just be coincident.